src/Home.jsx

```diff
@@ -0,0 +1,106 @@
+
+import { useState, useEffect } from "react";
+import Modal from "./components/Modal.jsx";
+import Country from "./components/Country.jsx";
+import Header from "./components/Header.jsx";
+import RegionFilter from "./components/RegionFilter.jsx";
+import SearchBox from "./components/SearchBox.jsx";
+
+const Home = ({ theme, toggleTheme }) => {
+    const [countries, setCountries] = useState([]);
+    const [allCountries, setAllCountries] = useState([]);
+    const [selectedCountry, setSelectedCountry] = useState(null);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        fetch("https://restcountries.com/v3.1/all")
+            .then((response) => {
```

Comment:
Hardcoded strings like this should be placed inside constants folder. 

Another thing - it is recommended to take all of the logic that handles something and extract it to service/ under the api folder.